SalesforceCommerceCloud/pwa-kit
# Add Basic URL Customization Support

# Description

As an initial step to provide multi site support for the PWA, this PR implemented the url customization feature on the PWA KIT to allow developers to customize their url params for locale and site when they'd like to. The approach follows the result of from the [spike doc](https://salesforce.quip.com/4bAqAZxu0PC0), which is to have a utility function to handle routes path, and a utility to build the path for each navigation across the app.

The url customization will allow developers to customize their url param as a `path` or as a `query_param`. Currently, it allows users to customization the **locale** and **siteAlias**.  

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR
- Run the app
- With `locale` set as `path` in the customization, you should be able to see the locale should be part of the url when you navigate through the app
- To test the `locale` as query_param, change your setting inside `pwa-kit-config.json`, reload the page, you should see the locale should be in the url as a query_param as opposed to be part of the path
- Changing locale in the footer should work as expected

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
